mobile/src/components/courses/courses.tsx

```diff
@@ -15,21 +17,37 @@ import {
   useAppNavigate,
   useAppSelector,
   useCallback,
+  useEffect,
   useFocusEffect,
   useState,
 } from '~/hooks/hooks';
-import { coursesActions } from '~/store/actions';
+import { categoryActions, coursesActions } from '~/store/actions';
 
+import { CategoryList } from './components/category-list/category-list';
 import { styles } from './styles';
 
 const Courses: FC = (): ReactElement => {
   const [isLoading] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
+  const [activeCategoryId, setActiveCategoryId] = useState<number | null>(null);
```

Comment:
We don't need to store this locally and in the store at the same time. Choose only one option